code/modules/shuttles/shuttle_supply_multi.dm

```diff
@@ -0,0 +1,825 @@
+		var/check2 = (get_step(T, EAST).density ? TRUE : FALSE)
+		if(check1 && check2)
+			in_elevator_controller = new/obj/structure/elevator_control_mount(T)
+			break
+
+/datum/shuttle/ferry/supply/multi/pick_loc()
+	RETURN_TYPE(/turf)
+	if(!supply_controller.multi_fakezlevel_elevator)
+		return GLOB.supply_elevator_turfs[target_zlevel]
+
+/datum/shuttle/ferry/supply/multi/get_location_area(location_id)
+	if(moving_status != SHUTTLE_INTRANSIT)
+		if(!fake_zlevel)
+			return area_offsite
+		else
+			return get_area(GLOB.supply_elevator_turfs[fake_zlevel])
+	else
+		return shaft_transit_area
+
+/datum/shuttle/ferry/supply/multi/proc/get_movement_direction()
+	//0 means DOWN; 1 means UP
+	if(moving_status == SHUTTLE_IDLE)
+		return null
+
+	if(!target_zlevel)
+		return 0
+
+	else if(!fake_zlevel)
+		return 1
+
+	else if(target_zlevel < fake_zlevel)
+		return 1
+	else
+		return 0
+
+/datum/shuttle/ferry/supply/multi/at_station()
+	return fake_zlevel
+
+/datum/shuttle/ferry/supply/multi/short_jump(area/origin, area/destination)
+	if(moving_status != SHUTTLE_IDLE)
+		return
+
+	if(isnull(location) || target_zlevel == fake_zlevel)
+		return
+
+	recharging = 1
+
+	if(!destination)
+		if(!target_zlevel)
+			destination = area_offsite
+		else
+			destination = get_area(GLOB.supply_elevator_turfs[target_zlevel])
+	if(!origin)
+		if(!fake_zlevel)
+			origin = area_offsite
+		else
+			origin = get_location_area()
+
+	moving_status = SHUTTLE_WARMUP
+	update_controller()
+	if(!target_zlevel && fake_zlevel)
+		playsound(GLOB.supply_elevator_turfs[fake_zlevel], 'sound/machines/cargo_alarm.ogg', 50, 0) //ELEVATOR IS GOING TO ASRS GERT UFF!!
+		raise_railingz(fake_zlevel)
+
+	sleep(warmup_time)
+
+	if (moving_status == SHUTTLE_IDLE)
+		return	//someone cancelled the launch
+
+	raise_railingz(fake_zlevel)
+
+	if (!target_zlevel)
+		sleep(12)
+		if(forbidden_atoms_check())
+			//cancel the launch because of forbidden atoms. announce over supply channel?
+			moving_status = SHUTTLE_IDLE
+			playsound(GLOB.supply_elevator_turfs[fake_zlevel], 'sound/machines/buzz-two.ogg', 50, 0)
+			if(fake_zlevel == supply_controller.primary_fake_zlevel)
+				lower_railingz(fake_zlevel)
+			update_controller()
+			return
+	else if(!fake_zlevel)	//at ASRS
+		supply_controller.buy()
+
+	if(target_zlevel)
+		moving_status = SHUTTLE_IDLE
+		for(var/turf/T in shaft_transit_area)
+			elevator_animationz[target_zlevel].vis_contents += T
+		moving_status = SHUTTLE_WARMUP
+	if(fake_zlevel)
+		moving_status = SHUTTLE_IDLE
+		for(var/turf/T in shaft_transit_area)
+			elevator_animationz[fake_zlevel].vis_contents += T
+		moving_status = SHUTTLE_WARMUP
+
+	//We pretend it's a long_jump by making the shuttle move to transit location for the "in-transit" period.
+	move_elevator_to_transit(origin, shaft_transit_area)
+
+	var/timing_multiplier = (target_zlevel > fake_zlevel) ? target_zlevel - fake_zlevel : fake_zlevel - target_zlevel
+	sleep(5 SECONDS * timing_multiplier) //muh transit time
```

Comment:
timer 🥺 in this case it's just calling move_elevator_to_destination after 5 SECONDS anyway